pkg/download/downloader.go

```diff
@@ -125,6 +125,18 @@ func extractTARGZ(targetDir string, at io.ReaderAt, size int64) error {
 			if err := os.MkdirAll(path, os.FileMode(hdr.Mode)); err != nil {
 				return errors.Wrap(err, "failed to create directory from tar")
 			}
+		case tar.TypeSymlink:
+			// the following two guard enforce a very conservative symlink policy:
+			// 1) no ".." references; and 2) no absolute path references
+			if strings.Contains(hdr.Linkname, "..") {
```

Comment:
this is... dangerous, and not necessarily authoritative, as one might work around it using escape characters etc. :) 

We should calculate it using a method we have that's tested, like IsSubPath or something.

Reply:
Also `..` references shouldn't be prevented. It makes sense to me have symlinks that refer to something on an upper directory that's still in the plugin's extract directory. (In fact, I assumed that's your primary use case.)

We need to probably calculate the final path of the link and use path cleaning + then check if it falls within the plugin's extraction directory.

We might consider linking the symlinks as absolute, but I don't see any upsides to that and makes it harder to move ~/.krew, if needed for some reason.

Reply:
are we ok with tarballs that escape to the staging directory? i.e. one level up? given that krew unpacks everything in a staging directory this is probably ok? if we aren't ok with this, then the escape detection logic gets more complex. we'd have to keep track of every tree in the forest being extracted

Reply:
these two sound like conflicting statements:

> are we ok with tarballs that escape to the staging directory?

> given that krew unpacks everything in a staging directory this is probably ok?

We need to make sure the computed target (path join+clean) of a symlink doesn't escape the `targetDir` parameter of the function.

Reply:
thanks, yes. the latest code does this. i forgot to update the comment. fixing.